nodejs/helpers.js

```diff
@@ -0,0 +1,174 @@
+const fs = require("fs")
+const mongodb = require("mongodb")
+const { ClientEncryption } = require("mongodb-client-encryption")(mongodb)
+const { MongoClient, Binary } = mongodb
+
+module.exports = {
+  readMasterKey: function(path = "./master-key.txt") {
+    return fs.readFileSync(path)
+  },
+  ClientBuilder: class {
+    constructor({
+      kmsProviders = null,
+      keyAltNames = "demo-data-key",
+      keyDB = "encryption",
+      keyColl = "__keyVault",
+      schema = null,
+      connectionString = "mongodb://localhost:27017",
+      mongocryptdBypassSpawn = false,
+      mongocryptdSpawnPath = "mongocryptd"
+    } = {}) {
+      if (kmsProviders === null) {
+        throw new Error("kmsProviders is required")
+      }
+      this.kmsProviders = kmsProviders
+      this.keyAltNames = keyAltNames
+      this.keyDB = keyDB
+      this.keyColl = keyColl
+      this.keyVaultNamespace = `${keyDB}.${keyColl}`
+      this.schema = schema
+      this.connectionString = connectionString
+      this.mongocryptdBypassSpawn = mongocryptdBypassSpawn
+      this.mongocryptdSpawnPath = mongocryptdSpawnPath
+      this.regularClient = null
+      this.csfleClient = null
+    }
+
+    /**
+     *
+     * @param {MongoClient} client
+     */
+    async ensureUniqueIndexOnKeyVault(client) {
+      try {
+        await client
+          .db(this.keyDB)
+          .collection(this.keyColl)
+          .createIndex("keyAltNames", {
+            unique: true,
+            partialFilterExpression: {
+              keyAltNames: {
+                $exists: true
+              }
+            }
+          })
+      } catch (e) {
+        console.error(e)
+        process.exit(1)
+      }
+    }
+
+    /**
+     * In the guide, https://docs.mongodb.com/ecosystem/use-cases/client-side-field-level-encryption-guide/,
+     * we create the data key and then show that it is created by
+     * using a find_one query. Here, in implementation, we only create the key if
```

Comment:
Nitpick: would this be less ambiguous if it specified how to prove it was created? e.g.: "show that it is created by retrieving it using a find_one query"